[Draft] Update cusparselt to 0.6.3.2 version (0.6.3 for short).

nWEIdia opened the pull request:
As of 01/24/2025, the nightly binary features 0.6.3.2 but the OSS CI seems to be still installing cusparselt 0.6.2 
This PR updates it to be consistent so that OSS CI also is installing cusparselt 0.6.3.2  

cc @ptrblck @eqy @tinglvv  @atalman @malfet

nWEIdia committed:
Update cusparselt to 0.6.3.2 version (0.6.3 for short).
Grace Hopper would like this version much better!

nWEIdia commented:
Hi @atalman , could you please help upload the new cusparselt package? nvidia-cusparselt-cu12==0.6.3 
Thanks!

jcaip commented:
@nWEIdia would it be possible to try and modify the build system to use the pypi package instead? I think this will make the upgrade process easier in the long run.

nWEIdia commented:
> @nWEIdia would it be possible to try and modify the build system to use the pypi package instead? I think this will make the upgrade process easier in the long run. #137374 for reference

@jcaip https://github.com/pytorch/pytorch/pull/137653/files was already merged to partially resolve the issue. The remaining work is to merge https://github.com/pytorch/builder/pull/2010. After that, the build system should be using pypi package instead of bundling the libcusparseLt.so with pytorch binaries.

jcaip commented:
@nWEIdia OK sounds good - thanks for updating to use the pip package!

nWEIdia commented:
@pytorchbot rebase

nWEIdia commented:
@pytorchbot merge -i

nWEIdia commented:
@pytorchbot rebase

nWEIdia committed:
Changed cusparselt version assert from 0.6.2 to 0.6.3

nWEIdia committed:
Run .github/regenerate.sh
Remove cusparselt 0.5.2 from cu121 because the binary depends on pypi
and 0.5.2 is not available. Using 0.6.3 nvidia-cusparselt-cu12
Resolving conflicts due to removing cu121

nWEIdia committed:
install_cusparselt.sh script using 0.6.3.2 as well

nWEIdia commented:
cc @eqy and @jcaip for help with https://github.com/pytorch/pytorch/actions/runs/11689439941/job/32555853937 failures, thanks!

nWEIdia commented:
Linking with https://github.com/pytorch/pytorch/pull/137427 and cc @j4yan as fyi.

jcaip commented:
> cc @eqy and @jcaip for help with https://github.com/pytorch/pytorch/actions/runs/11689439941/job/32555853937 failures, thanks!

Thanks @nWEIdia, I think some of https://github.com/pytorch/pytorch/pull/137427 needs to land, I'll pull the cuSPARSELt update part out and land that first.

nWEIdia commented:
Thanks @jcaip for the pointer!  I will keep an eye on #137427's landing.

nWEIdia commented:
@pytorchbot rebase
